packages/app-shell-shared/src/utils/conditionUtils.ts

```diff
@@ -0,0 +1,144 @@
+import type {
+  ConditionMetadata,
+  HvAppShellConditionConfig,
+} from "../types/condition";
+
+export const processElementConditions = (
+  conditions: HvAppShellConditionConfig[] | undefined,
+  elementPath: string,
+  elementType: ConditionMetadata["elementType"],
+  elementKey: string | number,
+  hookIndexRef: { current: number },
+  metadata: ConditionMetadata[],
+): void => {
+  if (!conditions) {
+    return;
+  }
+
+  for (let condIdx = 0; condIdx < conditions.length; condIdx++) {
+    const cond = conditions[condIdx];
+    metadata.push({
+      hookIndex: hookIndexRef.current++,
+      path: `${elementPath}.conditions[${condIdx}]`,
+      bundle: cond.bundle,
+      elementType,
+      elementPath,
+      elementKey,
+    });
+  }
+};
+
+/**
+ * Reconcile filtered configs to preserve object references when content hasn't changed
```

Comment:
Last time we discussed this, reconciliation would be done as part of filtering, given the previous filtered value.